modules/experimental/src/gpgpu/point-in-polygon/gpu-point-in-polygon.js

```diff
@@ -0,0 +1,183 @@
+import GL from '@luma.gl/constants';
+import {Buffer, Texture2D, assert} from '@luma.gl/webgl';
+import {isWebGL2} from '@luma.gl/gltools';
+import {Transform} from '@luma.gl/engine';
+import {default as textureFilterModule} from './texture-filter';
+import {POLY_TEX_VS, FILTER_VS} from './shaders';
+const TEXTURE_SIZE = 512;
+import * as Polygon from './polygon';
+
+export default class GPUPointInPolygon {
+  constructor(gl, opts = {}) {
+    this.gl = gl;
+    assert(isWebGL2(gl)); // supports WebGL2 only
+    this.textureSize = TEXTURE_SIZE;
+    this._setupResources();
+    this.update(opts);
+  }
+
+  update({polygons, textureSize} = {}) {
+    if (textureSize) {
+      this.textureSize = textureSize;
+    }
+    if (!polygons || polygons.length === 0) {
+      return;
+    }
+
+    const {vertices, indices, vertexCount, ids} = triangulatePolygons(polygons);
+    this._updateResources(vertices, indices, ids, vertexCount);
+  }
+
+  filter({positionBuffer, filterValueIndexBuffer, count}) {
+    this.filterTransform.update({
+      sourceBuffers: {
+        a_position: positionBuffer
+      },
+      feedbackBuffers: {
+        filterValueIndex: filterValueIndexBuffer
+      },
+      elementCount: count
+    });
+    const {polygonTexture, boundingBox} = this;
+
+    this.filterTransform.run({
+      moduleSettings: {boundingBox, texture: polygonTexture}
+    });
+  }
+
+  // PRIVATE
+
+  _setupResources() {
+    const {gl} = this;
+
+    // texture to render polygons to
+    this.polygonTexture = new Texture2D(gl, {
+      format: GL.RGB,
+      type: GL.UNSIGNED_BYTE,
```

Comment:
So a limit of 255 polygons? Probably fine, but maybe an assert somewhere would be worthwhile.

Reply:
Updated the docs to clarify this.